EE.C

```diff
@@ -0,0 +1,478 @@
+                float SBM2 = 0.;
+                if(abs(lep1_jet1_mass - lep2_jet2_mass) < abs(lep2_jet1_mass - lep1_jet2_mass)){
+                    SBM1 = lep1_jet1_mass;
+                    SBM2 = lep2_jet2_mass;
+                }
+                else{
+                    SBM1 = lep2_jet1_mass;
+                    SBM2 = lep1_jet2_mass;
+                }
+
+                //Fill max SBM
+                //if(SBM1>170. || SBM2>170.){//>170 cut
+                    if(SBM1<SBM2)SBM_DY->Fill(SBM2);
+                    else SBM_DY->Fill(SBM1);
+
+
+
+                    //METvsMmm
+                    //if(Nnon_bjet == 1 && Nbjet == 0){
+                    float Mmm = ((lep1)+(lep2)).M();
+                    float MET_t = MET[0];
+                    float METvsMmm = MET_t/Mmm ;
+                    //if(METvsMmm<0.2){ // <0.2 cut
+                        METvsMmm_DY->Fill(METvsMmm);
+
+
+                        MET_DY->Fill(MET_t);
+                        dilep_mass_DY->Fill(Mmm);
+                        //}
+                        //HT-LT
+                        float DHTLT = -999.;
+                        if(Nbjet == 2){
+                            DHTLT = (bJet_pt[0]+bJet_pt[1]) - (Lep_pt[0]+Lep_pt[1]);
+                        }
+                        else if(Nbjet == 1){
+                            DHTLT = (bJet_pt[0]+Jet2_nonB_PT) - (Lep_pt[0]+Lep_pt[1]);
+                        }
+                        else continue;
+
+                        //if(DHTLT<0){//<0 cut
+                            HTLT_DY->Fill(DHTLT);
+                        //}
+                //}
+                //}
+                float Mm_1 = (lep1).M();
+                float Mm_2 = (lep2).M();
+                Mass_DY->Fill(Mm_1);
+                Mass_DY->Fill(Mm_2);
+            }
+            //leading jet & 2nd leading jet pT    
+            if(secjet_pT == -999) secjet_pT = leadjet_pT;
+            LeadJetpT_DY->Fill(leadjet_pT);
+            SecJetpT_DY->Fill(secjet_pT);
+
+	  }
+      }
+      }
+      }
+      
+      
+   }
+  
+   run_DY->Write();
+
+   leppT_DY->Write();
+   elepT_DY->Write();
+   mupT_DY->Write();
+   
+   lepeta_DY->Write();
+   eleEta_DY->Write();
+   muEta_DY->Write();
+
+   lepphi_DY->Write();
+   elePhi_DY->Write();
+   muPhi_DY->Write();
+
+   dPhi_DY->Write();
+   dR_DY->Write();
+   ee_mass_DY->Write();
+
+   eleIso_DY->Write();
+   muIso_DY->Write();
+
+   jetpT_DY->Write();
+   jeteta_DY->Write();
+
+   bjetpT_DY->Write();
+   bjeteta_DY->Write();
+   
+   njvsnbj_DY->Write();
+   //bjetpT_dR_DY->Write();
+   //bjetpT_elepT_DY->Write();
+   //bjetpT_mupT_DY->Write();
+
+   LeadJetpT_DY->Write();
+   SecJetpT_DY->Write();
+
+//   btagEvent_DY->Write();
+
+   njet_DY->Write();
+   nbjet_DY->Write();
+   id_DY->Write();
+   
+   MET_DY->Write();
+   dilep_mass_DY->Write();
+
+   SBM_DY->Write();
+   METvsMmm_DY->Write();
+   HTLT_DY->Write();
+
+   Mass_DY->Write();
+
+   output->Close();
+
+}
+
```

Comment:
In addition to @rathjd comments:

- **source files should be under src folder or one some occasions under test, but never in the root of the repository**. Please move `EE.c` and `EE.h` with the `git mv` command
- I would also rename them to some more sounding name, like DiEleHistograms or so
- Please fix the indentations in the file and remove the extra blank lines
- Please provide more detailed comments on what each block of code is doing. A generic description of what this file is doing and how to use it is also required